<a href="https://colab.research.google.com/github/ArtaviousA1/The-python-Assignments/blob/main/week_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# IMPORT STATEMENTS
# 1. Import math

# FUNCTION DEFINITIONS
# 1. Define area(radius): returns pi * r squared
# 2. Define circumference(radius): returns 2 * pi * r
# 3. Define volume(radius, height):
#    - Call area(radius) and multiply by height.
#    - Return the result.

# PROGRAM LOGIC
# 1. Input for radius (float)
# 2. Call and print area and circumference.
# 3. Input for height (float)
# 4. Call and print volume.
import math

def area(radius):
    return math.pi * (radius ** 2)

def circumference(radius):
    return 2 * math.pi * radius

# Get user input for radius
radius_input = float(input("Enter the radius of the cylinder: "))

# Calculate and print area and circumference
calculated_area = area(radius_input)
calculated_circumference = circumference(radius_input)

print(f"Area: {calculated_area}")
print(f"Circumference: {calculated_circumference}")



# Task
Define Python functions `area(radius)`, `circumference(radius)`, and `volume(radius, height)` to calculate the area of a circle, the circumference of a circle, and the volume of a cylinder, respectively. The `volume` function should utilize the `area` function. Then, prompt the user for the radius and height of a cylinder, and print the calculated area, circumference, and volume.

## Define and Test Area and Circumference

### Subtask:
Define the `area(radius)` and `circumference(radius)` functions. Collect user input for the radius as a float. Call these functions with the input radius and print the results to verify accuracy as per the example provided.


**Reasoning**:
The subtask requires defining the `area` and `circumference` functions, taking user input for the radius, and then calling these functions to print the results. This can be achieved with a single Python code block.



In [1]:
import math

def area(radius):
    return math.pi * (radius ** 2)

def circumference(radius):
    return 2 * math.pi * radius

# Get user input for radius
radius_input = float(input("Enter the radius of the cylinder: "))

# Calculate and print area and circumference
calculated_area = area(radius_input)
calculated_circumference = circumference(radius_input)

print(f"Area: {calculated_area}")
print(f"Circumference: {calculated_circumference}")

Enter the radius of the cylinder: 5
Area: 78.53981633974483
Circumference: 31.41592653589793


# Task
Define the `volume(radius, height)` function, which should call the `area(radius)` function internally. Collect user input for the height as a float. Call the `volume` function with the previously entered radius and the new height, and print the result to verify accuracy.

## Define and Test Volume

### Subtask:
Define the `volume(radius, height)` function, which should call the `area(radius)` function internally. Collect user input for the height as a float. Call the `volume` function with the previously entered radius and the new height, and print the result to verify accuracy.


**Reasoning**:
The subtask requires defining the `volume` function, taking user input for the height, and then calling the `volume` function with the previously obtained radius and the new height to print the result. This can be achieved with a single Python code block by extending the existing code.



In [2]:
import math

def area(radius):
    return math.pi * (radius ** 2)

def circumference(radius):
    return 2 * math.pi * radius

def volume(radius, height):
    return area(radius) * height

# Get user input for radius
radius_input = float(input("Enter the radius of the cylinder: "))

# Calculate and print area and circumference
calculated_area = area(radius_input)
calculated_circumference = circumference(radius_input)

print(f"Area: {calculated_area}")
print(f"Circumference: {calculated_circumference}")

# Get user input for height
height_input = float(input("Enter the height of the cylinder: "))

# Calculate and print volume
calculated_volume = volume(radius_input, height_input)
print(f"Volume: {calculated_volume}")

Enter the radius of the cylinder: 5
Area: 78.53981633974483
Circumference: 31.41592653589793
Enter the height of the cylinder: 10
Volume: 785.3981633974483


## Final Task

### Subtask:
Summarize the calculations performed, including the area, circumference, and volume, based on the user's input values.


## Summary:

### Q&A
*   The area for the given radius (implied as 5.0) was calculated as approximately 78.54.
*   The circumference for the given radius (implied as 5.0) was calculated. However, its specific value was not explicitly provided in the summary of findings from the code execution.
*   The volume for the given radius (implied as 5.0) and height (implied as 10.0) was calculated as approximately 785.40.

### Data Analysis Key Findings
*   The `volume(radius, height)` function was successfully defined, internally calling the `area(radius)` function to compute the base area.
*   User input was collected for both the radius and height of the cylinder.
*   For an implied radius of 5, the calculated area was approximately 78.5398.
*   For an implied radius of 5 and an implied height of 10, the calculated volume was approximately 785.3982.

### Insights or Next Steps
*   The implemented functions provide a functional and verifiable method for calculating the area, circumference, and volume of a cylinder given its radius and height.
*   These functions can be readily integrated into larger applications requiring geometric calculations for cylindrical shapes.
